irods/test/meta_test.py

```diff
@@ -646,6 +647,27 @@ def test_thoroughgoing_correctness_of_genquery_result_column_mappings__issue_642
         finally:
             del test_coll.metadata[meta_key]
 
+    def test_xml_mode_addresses_odd_metadata_characters__issue_582(self):
+        session = self.sess
+        try:
+            hexvalue = "53:6b:6f:70:65:43:61:6c:58:c3:af:c2:bf:c2:bd:c3:af:c2:bf:c2:bd:23:01:32:64:31"
+            hex_list = [int(byte, 16) for byte in hexvalue.split(":")]
+            string_value = "".join([chr(byte) for byte in hex_list])
+
+            try:
+                from irods.helpers import xml_mode
+            except ImportError:
+                # Create an xml_mode context manager that is a no-op.
+                def _dummy(*x): yield
+                xml_mode = contextlib.contextmanager(_dummy)
```

Comment:
Why does this need to be wrapped in a try-catch block?

Reply:
xml_mode is either importable or not, so if not , we make an alternative definition

Reply:
Therfore if xml_mode doesn't exist, the test is guaranteed to fail for the right reason (that xml_mode did nothing of note, rather than that it failed to import)

Reply:
Hmm, when would it not be importable?

Reply:
The only case in which it wouldn't be importable is: running a PRC version before xml_mode was introduced to the library.  I'd been using this code in a script to demonstrate to myself that things failed when xml_mode was just a no-op.

So in retrospect, you are right, I should take this out.  It's not a valid part of the test.

Reply:
also done.